### PR TITLE
changes to get usb working on a L1. Adds a usb_serial example too

### PR DESCRIPTION
**Pull request by shufps**
the USB implementation didn't enable the internal USB Pull-Up of the L1s.

Additional setting the Alternate Functions messes up USB and it doesn't work anymore.

PR disables the setting for the L1s buuuuut now warnings are generated and I don't know how to fix them :pleading_face:

### CODE REVIEW

File: examples/stm32l1/Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,7 @@ embassy-sync = { version = "0.5.0", path = "../../embassy-sync", features = ["de
 embassy-executor = { version = "0.5.0", path = "../../embassy-executor", features = ["arch-cortex-m", "executor-thread", "defmt", "integrated-timers"] }
 embassy-time = { version = "0.3.0", path = "../../embassy-time", features = ["defmt", "defmt-timestamp-uptime", "tick-hz-32_768"] }
 embassy-stm32 = { version = "0.1.0", path = "../../embassy-stm32", features = [ "defmt", "stm32l151cb-a", "time-driver-any", "memory-x"]  }
+embassy-usb = { version = "0.1.0", path = "../../embassy-usb", features = ["defmt"] }
```
Comment (Dirbaio) [inline examples/stm32l1/Cargo.toml:12]: perhaps you forgot to `git add` the usb example file?

Comment (shufps) [reply examples/stm32l1/Cargo.toml:12]: yes, thx - I added it :blush:

**Review by Dirbaio — APPROVED** [review]
thank you!

**Comment by Dirbaio** [comment]
bender run

**Comment by Dirbaio** [comment]
seems CI needs some fixes

**Comment by shufps** [comment]
> seems CI needs some fixes

yes, that is what I mentioned in the PR - the AF and the dm and dp pins are now not used anymore and it generates warnings.

Unfortunately I have no idea how to fix this in a clean way :disappointed:

File: embassy-stm32/src/usb/usb.rs [hunk 2]
```diff
@@ -287,10 +286,12 @@ impl<'d, T: Instance> Driver<'d, T> {
...
-        dp.set_as_af(dp.af_num(), AFType::OutputPushPull);
-
-        #[cfg(not(stm32l1))]
-        dm.set_as_af(dm.af_num(), AFType::OutputPushPull);
+        {
+            dp.set_as_af(dp.af_num(), crate::gpio::sealed::AFType::OutputPushPull);
+            dm.set_as_af(dm.af_num(), crate::gpio::sealed::AFType::OutputPushPull);
+        }
+        #[cfg(stm32l1)]
+        let _ = (dp, dm); // suppress "unused" warnings.
```
Comment (shufps) [inline embassy-stm32/src/usb/usb.rs:294]: oh nice :exploding_head:

**Comment by shufps** [comment]
There is an interesting thread about this: https://community.st.com/t5/stm32-mcus-products/stm32l151-usb-what-actually-switches-pa11-pa12-from-gpio-to-usb/td-p/550560

tl;dr: People describing that they never found any code for the L1 that sets AF pins for USB and if they try USB doesn't work anymore ... lol :see_no_evil:
